src/main/java/seedu/duke/command/Command.java

```diff
@@ -0,0 +1,88 @@
+package seedu.duke.command;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Superclass for all supported commands in Duke.<br>
+ *
+ * Each Command is initialised with the arguments issued
+ *     by the user. Execute the specified Command by calling
+ *     execute().<br>
+ *
+ * Child classes must provide the static isValidCommand() method for checking whether a set of
+ *     arguments are valid for a given command.
+ */
+public abstract class Command {
+    private static final String ARGUMENT_DELIMITER = "--";
+    private static final String DELIMITER_FOR_WORDS = " ";
+    private final HashMap<String, String> arguments;
```

Comment:
Might be good to write what the key and value for the Hashmap are  (argument name, description) as it may not be immediately obvious to the reader